### PR TITLE
Add Jobs guide: Process Large Datasets

### PR DESCRIPTION
**Pull request by davanstrien**
## What

Adds a new Jobs documentation page — **Process Large Datasets** (`docs/hub/jobs-large-datasets.md`) — under the Jobs section of the nav, plus an inbound pointer from `jobs-configuration`'s Volumes section.

It's a task-oriented guide for working with datasets **larger than a Job's ephemeral disk**, opening with a short decision rule and then covering each approach:

- **Stream** with `datasets` (`streaming=True`) — links the [Stream guide](https://huggingface.co/docs/datasets/stream) and the [100× more efficient](https://huggingface.co/blog/streaming-datasets) blog
- **Read & filter over `hf://`** directly with Polars/DuckDB/pandas (native readers, no mount)
- **Mount** a dataset/model/bucket for tools that expect local file paths
- **Save results** to a Storage Bucket (DuckDB `COPY` straight from `hf://` to the mount) or `push_to_hub`
- A **Common Crawl** worked example: stream a WET shard straight from `hf://` → DuckDB, no download

It links the canonical reference pages (`jobs-pricing`, `jobs-configuration#volumes`, `storage-buckets-access`, `datasets/stream`, per-library dataset pages) rather than restating them.

## Verification

Every snippet was run against live Jobs. Timings that shaped the page (same aggregate query, same ~28 GB / 14-file Parquet glob):

| Read path | Time |
|---|---|
| Polars native `hf://` scan | **~4 min** (in the doc) |
| DuckDB native `hf://` | ~18 min |
| DuckDB via `register_filesystem(HfFileSystem())` | did not finish (killed at 30-min default timeout) |
| Polars over a `-v` repo mount | ~6 min *per file* (~85 min extrapolated) |

Hence the page's structure: `hf://` native readers for scans, mounts for whole-file/local-path access, explicit `--timeout` guidance for network-bound scans. The Common Crawl example runs verbatim in ~1 min on the default CPU flavor with no token; its printed output is included on the page.

## Notes

- #2554 (jobs-serving) touched the same `_toctree.yml` region and has since merged; this branch is mergeable and rebases cleanly on `main` (no conflicts).

---

> [!NOTE]
> **Low Risk**
> Documentation-only changes with no runtime or API impact.
> 
> **Overview**
> Adds a new Jobs doc page **Process Large Datasets** and registers it in the Hub nav (after Examples & Tutorials).
> 
> The guide explains how to work with data bigger than a Job’s ephemeral disk: a short “which approach?” decision tree, then sections on **streaming** (`datasets` + Spark), **native `hf://` scans** (Polars/DuckDB/pandas) with `--timeout` and parallel-job guidance, **volume mounts** for path-based tools, **persisting output** to Storage Buckets or `push_to_hub`, and a **Common Crawl WET** end-to-end example. It points to existing pricing, configuration, bucket, and per-library docs instead of duplicating them.
> 
> **Configuration** gains a Volumes tip that lazy mounts enable oversized datasets and links to this page.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 7cd203e257e683523c46a6419a7069b723b61b62. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by davanstrien** [comment]
IMO, we can keep iterating on this as libraries mature, but nice to document some patterns already.

File: docs/hub/jobs-large-datasets.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+Common Crawl mirrors its archive to the bucket [`commoncrawl/commoncrawl`](https://huggingface.co/buckets/commoncrawl/commoncrawl) — hundreds of TB. Stream one WET (plaintext) shard straight from `hf://`, parse it, and query it with DuckDB; only a few MB transit because the gzip is read sequentially and stopped early:
+
+```python
+# /// script
+# requires-python = ">=3.11"
+# dependencies = ["huggingface_hub>=1.0", "fastwarc>=0.15", "duckdb>=1.0"]
+# ///
+import duckdb
+from fastwarc.warc import ArchiveIterator, WarcRecordType
+from huggingface_hub import HfFileSystem
```
Comment (lhoestq) [inline docs/hub/jobs-large-datasets.md]: (minor)
```suggestion
from huggingface_hub import hffs
```

File: docs/hub/jobs-large-datasets.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+WET = (
+    "buckets/commoncrawl/commoncrawl/crawl-data/CC-MAIN-2026-17/"
+    "segments/1775805908305.14/wet/"
+    "CC-MAIN-20260410081153-20260410111153-00000.warc.wet.gz"
+)
+
+rows = []
+with HfFileSystem().open(WET, "rb") as f:
```
Comment (lhoestq) [inline docs/hub/jobs-large-datasets.md]:
```suggestion
with hffs.open(WET, "rb") as f:
```

File: docs/hub/jobs-large-datasets.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+```python
+from datasets import load_dataset
+
+ds = load_dataset("HuggingFaceFW/fineweb-edu", "sample-10BT", split="train", streaming=True)
+for example in ds.take(1000):
+    ...  # streams in as you iterate, nothing hits disk
+```
+
+A streamed dataset is an `IterableDataset` supporting lazy `.filter()`, `.map()`, `.shuffle(buffer_size=...)`, and `.batch()`, and can be passed straight to a PyTorch `DataLoader` or a `Trainer` to train on data larger than disk. See the [Stream guide](/docs/datasets/stream) for the full API, and [Examples & Tutorials](./jobs-examples) for end-to-end training walkthroughs.
+
```
Comment (lhoestq) [inline docs/hub/jobs-large-datasets.md:36]: could be worth mentioning distributed streaming as well

here with spark:

````suggestion

Dataset streaming is also available in distributed setups using Spark:

```python
import pyspark_huggingface

df = spark.read.format("huggingface").option("config", "sample-10BT").load("HuggingFaceFW/fineweb-edu")
```

The resulting Spark dataframe is distributed: each worker has a subset of files to stream from. See the [Spark documentation](https://huggingface.co/docs/hub/datasets-spark) to see more examples on how to read, write and efficiently filter rows and columns.

````

File: docs/hub/jobs-large-datasets.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+```python
+from datasets import load_dataset
+
+ds = load_dataset("HuggingFaceFW/fineweb-edu", "sample-10BT", split="train", streaming=True)
+for example in ds.take(1000):
+    ...  # streams in as you iterate, nothing hits disk
+```
+
+A streamed dataset is an `IterableDataset` supporting lazy `.filter()`, `.map()`, `.shuffle(buffer_size=...)`, and `.batch()`, and can be passed straight to a PyTorch `DataLoader` or a `Trainer` to train on data larger than disk. See the [Stream guide](/docs/datasets/stream) for the full API, and [Examples & Tutorials](./jobs-examples) for end-to-end training walkthroughs.
```
Comment (lhoestq) [inline docs/hub/jobs-large-datasets.md]: and here with `datasets` (linking to the docs is enough)

```suggestion
A streamed dataset is an `IterableDataset` supporting lazy `.filter()`, `.map()`, `.shuffle(buffer_size=...)`, and `.batch()`, and can be passed straight to a PyTorch `DataLoader` or a `Trainer` to train on data larger than disk. See the [Stream guide](/docs/datasets/stream) for the full API, and [Examples & Tutorials](./jobs-examples) for end-to-end training walkthroughs for single-node or distributed setups.
```

File: docs/hub/jobs-large-datasets.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
+# Process Large Datasets
+
+Every Job comes with a fixed amount of local disk, set by its [hardware flavor](./jobs-pricing#pricing) (the **Ephemeral Storage** column, also shown by `hf jobs hardware`). You don't need to fit a whole dataset on that disk to work with it: datasets and [Storage Buckets](./storage-buckets) can be read directly from the Hub — streamed, queried, or mounted — so a single Job can process data far larger than its disk. This page covers the options and when to reach for each.
+
+## Which approach?
+
+- **Fits on disk** → a plain `load_dataset(...)` works as-is (or just pick a bigger flavor — up to 1 TB of ephemeral disk).
+- **Iterating over rows, or training** → [stream](#stream-the-dataset) it.
```
Comment (lhoestq) [inline docs/hub/jobs-large-datasets.md]:
```suggestion
- **Iterating over rows, processing or training** → [stream](#stream-the-dataset) it.
```

**Review by lhoestq — APPROVED** [review]
lgtm ! I added some minor suggestions
